### DISCUSSION TITLE
Why the connection line looks like this?

### DISCUSSION THREAD

**Discussion by tpoxa** [bcakmakoglu/vue-flow#467] [Q&A] [answered]
Hey. I noticed that the connection line looks different when I drag it from left to right than the right to left using a horizontal layout.
Can it be easily changed?

Screenshot is taken from here https://vueflow.dev/examples/horizontal.html

<img width="349" alt="Screenshot 2022-12-03 at 19 33 16" src="https://user-images.githubusercontent.com/431546/205458624-7bf958f8-6a99-4510-91e1-8a87adf4470d.png">

Thanks. It's a really great project.

**Comment by bcakmakoglu** [comment, marked as answer]
Currently I'm not at home so I can't really check what's going on, I'll check it out tomorrow 😄
Thank you for reporting though!

**Reply by bcakmakoglu** [reply]
Ok so I misunderstood your question at first, I thought that's an edge that just didn't connect properly but I see that you were just dragging a connection line.

So, what is going on is that the connection line will orient itself depending on if you start from a `target` or `source` handle.
You can change the orientation by using a custom connection line and using `getBezierPath` and changing the direction of the `targetHandle` to left/right/top/bottom. That will tell the connection line in which direction it should orient itself while dragging.
By default it will just reverse from the position it started at basically, that's why it might look "strange" when you start from a target handle and go "backwards", if that makes sense 😄

**Reply by tpoxa** [reply]
Thanks, @bcakmakoglu, but do you have an example of that? I am already using the customEdge example with `getBezierPath` in it.

**Reply by bcakmakoglu** [reply]
A custom edge isn't the same as a custom connection line.
You can check the basic example for a custom connection line [here](https://vueflow.dev/examples/edges/connection-line.html).
Play around with it and try modifying the source/target handle positions when using `getBezierPath` and you should be seeing different orientations :)

**Reply by bcakmakoglu** [reply]
(the example isn't using `getBezierPath` btw, you'd have to add it but the idea is the same 😄)

**Reply by tpoxa** [reply]
I appreciate your help. Unfortunately, I don't understand specific lists of actions to make connections lines look the same when I drag it from source to target and from target to source.

**Reply by tpoxa** [reply]
But after the latest upgrade, I have much more problems than how the lines look like.

**Reply by bcakmakoglu** [reply]
Please check the Migration guide on the docs page :)

**Reply by bcakmakoglu** [reply]
I don't quite understand your first question, could you elaborate on that further? What exactly is causing you trouble?

**Reply by tpoxa** [reply]
> I don't quite understand your first question, could you elaborate on that further? What exactly is causing you trouble?

Just don't like how the connection like looks when it's being dragged from target to source. The screenshot displays one of the examples published online. You just need to grab the line from Node 5 and drag it to Node 3. The line curves inside like avoiding being connected. It looks weird to me.

**Reply by bcakmakoglu** [reply]
I know what it looks like but as I said, you can implement your own custom connection line that flips the direction based on what direction your pulling it.
Right now that's just the behavior of the default connection line.

**Reply by bcakmakoglu** [reply]
You might want to try this snippet if you have trouble getting the directions of your connection line right:

```ts
const path = computed(() =>
  getBezierPath({
    ...props,
    targetPosition: props.targetX < props.sourceX ? ('right' as Position) : ('left' as Position),
  }),
)
```

That should give you the alignment you're asking for if I'm not wrong :)

**Reply by bcakmakoglu** [reply]
@tpoxa btw, with the next minor release `1.14.0` the connection line will automatically draw to the opposite position, which should probably give you the result you were looking for to begin with 😄 
Let me know if that's better for you after you've upgraded

**Reply by tpoxa** [reply]
Hey @bcakmakoglu I like your latest updates. The connection line now looks the same great when I drag it from the source as when I drag it from the target. Also, the overall process of connecting edges now is more pleasant. Thanks!
